### PR TITLE
Update react from 16 to 18.2

### PR DESCRIPTION
**Pull request by LudvigHz**
# Description

Updates react from 17 to 18 :))

# Result

No visual changes.

# Testing

- [x] I have thoroughly tested my changes.

Both SSR and client-side rendering work great.

---

Resolves ABA-213

### CODE REVIEW

**Review by ivarnakken — APPROVED** [review]
:shipit:

**Review by erlingfn — APPROVED** [review]
LGTM!

**Comment by LudvigHz** [comment]
I believe I fixed all the hydration issues now so this should finally work :tm:

File: cypress/support/utils.js [hunk 1]
```diff
@@ -39,12 +39,8 @@ export const button = (buttonText) => cy.contains('button', buttonText);
 
 export const selectEditor = (name) =>
   name
-    ? cy
-        .wait(500)
-        .get(`[name="${name}"] div[data-slate-editor="true"]`)
-        .click()
-        .wait(500)
-    : cy.wait(500).get('div[data-slate-editor="true"]').click().wait(500);
```
Comment (ivarnakken) [inline cypress/support/utils.js:47]: Very sexy

File: package.json [hunk 2]
```diff
@@ -120,6 +120,7 @@
     "@babel/preset-env": "^7.20.2",
     "@babel/preset-react": "^7.18.6",
     "@babel/preset-typescript": "^7.21.0",
+    "@cfaester/enzyme-adapter-react-18": "^0.6.0",
```
Comment (ivarnakken) [inline package.json:123]: You forgot to remove `enzyme-adapter-react-16`, but okay

**Pull request merged**